ResearchKit/Common/ORKImageCaptureView.m

```diff
@@ -140,10 +157,18 @@ - (UIImage *)capturedImage {
 - (void)setError:(NSError * __nullable)error {
     _error = error;
     _headerView.alpha = error==nil ? 0 : 1;
-    _headerView.instructionLabel.text = [error.userInfo valueForKey:NSLocalizedDescriptionKey];
-    if (error && self.continueTitle) {
-        _continueSkipContainer.continueButtonItem.title = self.continueTitle;
-        _continueSkipContainer.skipButtonItem = nil;
+    _headerView.instructionLabel.text = error==nil ? nil : [error.userInfo valueForKey:NSLocalizedDescriptionKey];
+    
+    if (error) {
+        // Show skip, if available, and hide the template and continue/capture button
+        _previewView.templateImageHidden = YES;
+        _continueSkipContainer.continueButtonItem = nil;
+        _continueSkipContainer.skipButtonItem = _skipButtonItem;
+    } else {
+        // Reset to running state
+        _previewView.templateImageHidden = NO;
```

Comment:
Once the error is set, the layout changes to only show the error, and no longer show the ``previewView`` containing the captured image.  So, you'd like to see the code thats in ``setCapturedImage`` and ``setError`` that deals with what to change the buttons to moved into its own function.  And so that function would have the three cases: error, captured image non-nil, and captured image nil.  Is that right?